community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/pruning/LogPruningImpl.java

```diff
@@ -31,15 +35,41 @@
 public class LogPruningImpl implements LogPruning
 {
     private final Lock pruneLock = new ReentrantLock();
+    private final FileSystemAbstraction fs;
     private final LogPruneStrategy pruneStrategy;
+    private final PhysicalLogFiles logFiles;
     private final Log msgLog;
 
-    public LogPruningImpl( LogPruneStrategy pruneStrategy, LogProvider logProvider )
+    public LogPruningImpl( FileSystemAbstraction fs,
+                           LogPruneStrategy pruneStrategy,
+                           PhysicalLogFiles logFiles,
+                           LogProvider logProvider )
     {
+        this.fs = fs;
         this.pruneStrategy = pruneStrategy;
+        this.logFiles = logFiles;
         this.msgLog = logProvider.getLog( getClass() );
     }
 
+    private LongStream getLogVersionsToDelete( long upToVersion )
```

Comment:
Because this was a smaller change to the `LogPruneStrategy` implementations. The log files have to be visited in oldest-to-newest order, but the log version brackets and the threshold computation is currently done in newest-to-oldest order. So to do this, we'll have to rewrite `ThresholdBasedPruneStrategy`.

Reply:
hm..don't know how bit that change is especially taking into account that ThresholdBasedPruneStrategy was already changed to return stream instead. But can live with this stream ok.